### PR TITLE
Propose Guidelines for Creating and Executing Tests in Vanilla OS

### PR DESCRIPTION
**Pull request by mirkobrombin**

### CODE REVIEW

**Review by pietrodicaprio — APPROVED** [review]

**Review by axtloss — APPROVED** [review]

**Review by taukakao — APPROVED** [review]

File: TESTS_GUIDELINES.md
Comment (kbdharun) [inline TESTS_GUIDELINES.md]:
```suggestion
Manual testing includes usability tests, user interaction tests, and comprehensive scenario tests. It is crucial not to rely solely on automated tests, but also involve human insights to catch issues that automated tests may miss.
```

**Review by kbdharun — APPROVED** [review]
LGTM, after this minor suggestion.

File: TESTS_GUIDELINES.md
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]:
```suggestion
It is important that all maintainers are familiar with and have understood the [Vanilla OS contribution guidelines](CONTRIBUTING.md).
```

**Review by taukakao — APPROVED** [review]

File: TESTS_GUIDELINES.md
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]: What about "It is the responsibility of the maintainers to ensure the continuous integration (CI) of tests into their respective projects."?

File: TESTS_GUIDELINES.md
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]: > Continuous Integration (CI)

If the above suggestion is accepted, we could just use "CI"

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

File: TESTS_GUIDELINES.md
Comment (mirkobrombin) [reply TESTS_GUIDELINES.md]: Yeah I like your proposal

**Review by 89luca89 — APPROVED** [review]

**Review by TheEvilSkeleton — APPROVED** [review]

File: TESTS_GUIDELINES.md
Comment (matbme) [inline TESTS_GUIDELINES.md:31]: Not sure if this is the right place, but we should enforce formatters like gofmt (integrated into gopls) for Go and black (or ruff which checks for black's rules) for Python.

Comment (mirkobrombin) [reply TESTS_GUIDELINES.md:31]: Not sure if this should be in those guidelines or the general ones

Comment (matbme) [reply TESTS_GUIDELINES.md:31]: Maybe it's best to put in the general guidelines.

**Review by matbme — APPROVED** [review]

**Pull request merged**
